library/bignum_core.h

```diff
@@ -60,6 +60,57 @@ size_t mbedtls_mpi_core_bitlen( const mbedtls_mpi_uint *A, size_t A_limbs );
 void mbedtls_mpi_core_bigendian_to_host( mbedtls_mpi_uint *A,
                                          size_t A_limbs );
 
+/**
+ * \brief   Perform a safe conditional copy of MPI which doesn't reveal whether
+ *          the condition was true or not.
```

Comment:
should we say something about what happens if the parameters are aliased (and test it)? Same for the other functions.

Reply:
Good question. If we follow the [current draft of the general conventions](https://github.com/Mbed-TLS/mbedtls/pull/6373/files#r988216519), this would be aliasing between an (input-)output parameter and an input parameter, and therefore allowed. It's rather silly on the face of it: why would you write X := X — and the “it comes from some higher level and the intermediate level doesn't know” argument doesn't really apply because the intermediate level has to know that it can modify X. On the other hand, in case this does arise, it would be better to avoid surprises. And it's very easy to handle in the code (`if (X == A) return;` — works for swap as well). (As usual, we would only allow exact aliasing, and not arbitrary overlap.)